android/KMEA/app/src/main/java/com/tavultesoft/kmea/data/KeyboardController.java

```diff
@@ -0,0 +1,196 @@
+/**
+ * Copyright (C) 2020 SIL International. All rights reserved.
+ */
+
+package com.tavultesoft.kmea.data;
+
+import android.content.Context;
+import android.util.Log;
+
+import com.tavultesoft.kmea.JSONParser;
+import com.tavultesoft.kmea.data.Keyboard;
+import com.tavultesoft.kmea.KeyboardPickerActivity;
+import com.tavultesoft.kmea.util.FileUtils;
+import com.tavultesoft.kmea.util.MapCompat;
+import com.tavultesoft.kmea.KMManager;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.ObjectInputStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+public class KeyboardController {
+  public static final String TAG = "InstalledKBList";
+  public static final String KMFilename_Installed_KeyboardsList = "keyboards_list.json";
+
+  private static KeyboardController instance;
+
+  /**
+   * @return get or create shared instance.
+   */
+  public static KeyboardController getInstance() {
+    if (instance == null) {
+      createInstance();
+    }
+    return instance;
+  }
+
+  /**
+   * create singleton instance.
+   */
+  private synchronized static void createInstance()
+  {
+    if (instance != null) {
+      return;
+    }
+    instance = new KeyboardController();
+  }
+
+  private boolean isInitialized = false;
+  private List<Keyboard> list;
+
+  public synchronized  void initialize(Context context) {
+    if (isInitialized) {
+      return;
```

Comment:
Do we need to be aware of attempts to `initialize` more than once?

Reply:
I'm not sure we need to re-initialize.

The internal list will be in sync with the JSON file as we keep the calls to KeyboardPickerActivity.saveList(). (My other branch)

Reply:
Yep. More concerned with whether a second call to `initialize` would indicate a potential logic flaw. I'm not sure where it's being called from...

Reply:
k, I'll add some logging if `initialize` gets called again